website/utils/url_utils.ts

```diff
@@ -2,16 +2,56 @@ import FirebaseRequest, { DatasetMetaData } from "../../public/firebase";
 
 import { AppProps, GlobalViewerSettings } from "../../src/aics-image-viewer/components/App/types";
 import { ViewMode } from "../../src/aics-image-viewer/shared/enums";
-import { ViewerChannelSettings } from "../../src/aics-image-viewer/shared/utils/viewerChannelSettings";
+import {
+  ViewerChannelSetting,
+  ViewerChannelSettings,
+} from "../../src/aics-image-viewer/shared/utils/viewerChannelSettings";
 
-const paramKeys = ["mask", "ch", "luts", "colors", "url", "file", "dataset", "id", "view"];
-type ParamKeys = (typeof paramKeys)[number];
-type Params = { [_ in ParamKeys]?: string };
+const CHANNEL_STATE_KEY_REGEX = /^c[0-9]+$/;
+// Match colon-separated pairs of alphanumeric strings
+const LUT_REGEX = /^[a-z0-9.]*:[ ]*[a-z0-9.]*$/;
+const HEX_COLOR_REGEX = /^[0-9a-fA-F]{6}$/;
 
-function urlSearchParamsToParams(searchParams: URLSearchParams): Params {
+export type ViewerChannelSettingJson = {
+  /** Color, as a 6-digit hex color */
+  col?: string;
+  /** Colorize. "1" is enabled. */
+  clz?: "1" | "0";
+  /** Colorize alpha, in the [0, 1] range. */
+  cza?: string;
+  /** Opacity, in the [0, 1 range] */
+  isa?: string;
+  /** LUT to map from intensity to opacity. Should be two alphanumeric values separated
+   * by a colon, e.g. "0:255", "p50:p90", or "autoij:0".
+   */
+  lut?: string;
+  /** Volume enabled. "1" is enabled. */
+  ven?: "1" | "0";
+  /** Isosurface enabled. "1" is enabled. */
+  sen?: "1" | "0";
+  /** Isosurface value, in the [0, 255] range. */
+  isv?: string;
+};
+
+const baseParamKeys = ["mask", "url", "file", "dataset", "id", "view"] as const;
```

Comment:
seeing this, I am trying to remember what's the difference between file and url?  (Another documentation issue).
Dataset is an identifier for a Cell Feature Explorer dataset that is stored in our firebase db, and I believe id is an image id that's a member of dataset.

Reply:
It looks like `file` pulls directly from your file server? Currently, it appends the contents of `file` to `http://dev-aics-dtp-001.corp.alleninstitute.org/dan-data/` as a base URL.

The comments near it reads:
```
// quick way to load a atlas.json from a special directory.
//
// ?file=relative-path-to-atlas-on-isilon
```

Should I remove it?

Reply:
Thanks -- yes this seems like a reasonable time to remove it or put it in the deprecated list.  It's just an extra thing to maintain for no reason and there's no good reason why that url should be hardcoded into this app.